java/memory/memory-core/src/main/java/org/apache/arrow/memory/BufferLedger.java

```diff
@@ -178,7 +179,7 @@ public void retain() {
   @Override
   public void retain(int increment) {
     Preconditions.checkArgument(increment > 0, "retain(%s) argument is not positive", increment);
-    if (BaseAllocator.DEBUG) {
+    if (BaseAllocator.DEBUG && historicalLog != null) {
```

Comment:
nit: one more BaseAllocator.DEBUG we can remove!

Reply:
Thanks, changed.